src/Metadata/MetadataParser.php

```diff
@@ -0,0 +1,168 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\Image\Metadata;
+
+use Contao\Image\Exception\InvalidArgumentException;
+use Contao\Image\Exception\InvalidImageContainerException;
+use Contao\Image\Exception\InvalidImageMetadataException;
+
+/**
+ * @final
+ */
+class MetadataParser
```

Comment:
```suggestion
class MetadataReaderWriter
```
Better name for this class?